### PR TITLE
chore(ci): ci optimizations

### PR DESCRIPTION
**Pull request by g-cheishvili**
## Description
- Everything is moved to NX executors
- Now you can launch e2e tests per page using nx. e.g `nx run docs-core-action-bar:e2e` and it will launch `docs:serve:e2e` and test it right away. You can use `--baseUrl=http://localhost:4200` and it will test that URI, same goes for remote testing. Bear in mind that it is cacheable.
- You can launch whole application e2e tests via `docs:e2e-app:{affected|local}-{all|remote}` and optionally can pass `devServerTarget` and/or `baseUrl` as targets
- Standardized folder structure and naming across `platform` component docs
- In PRs it will launch only tests for pages that have changed.
- You now can not launch `core:build`, there was no use of that target for core, platform and fn. Now to build them you need to launch `{projectName}:build-umbrella`
- Generated a bunch of libraries for documentation pages. Every docs page = new doc library
- Examples are not loaded now into the final bundle, they have become `assets`. They are retrieved using fetch now, so we can in nearest future make them lazy-loaded.
- Release notes are now fixed.
- Create release script now is triggered through Github actions, not by script
- Updated nx to 14.5.10 while was at it
- Generation of typedocs are now cacheable
- Fixed a few occurrences of Circular dependencies in core
- Deleted all unneeded files from `e2e/wdio/baselineScreenshot`, from `scripts` folder, from `ci-scripts` folder
- Tidied up `package.json` scripts so we do not have that much unneeded scripts anymore

### CODE REVIEW

**Review by InnaAtanasova — APPROVED** [review]
took a few minutes for the browser to load the files :D

**Review by platon-rov — COMMENTED** [review]
awesome! left minor comments about items from my pr, also please fix the pipeline.

File: nx.json [hunk 1]
```diff
@@ -13,6 +13,7 @@
         "default": {
             "runner": "@nrwl/nx-cloud",
             "options": {
+                "accessToken": "MzFiYWRhMTItYTAxYi00MGZjLWJmNDEtOTllMzEwZjI5ODNkfHJlYWQ=",
```
[the remark this answers is not in the document]
Comment (N1XUS) [reply nx.json:16]: Should this actually be stored in the repo?

[the remark this answers is not in the document]
Comment (g-cheishvili) [reply nx.json:16]: It is read-only token, so it is okay

[the remark this answers is not in the document]
Comment (g-cheishvili) [reply nx.json:16]: It is NGX token, given by Deno

[the remark this answers is not in the document]
Comment (mikerodonnell89) [reply nx.json]: https://nx.dev/nx-cloud/account/access-tokens

File: apps/docs/project.json [hunk 2]
```diff
@@ -122,10 +222,10 @@
             },
             "outputs": ["{options.outputPath}"],
             "defaultConfiguration": "",
```
[the remark this answers is not in the document]
Comment (mikerodonnell89) [reply apps/docs/project.json:224]: @platon-rov can you open PRs for this?

File: apps/docs/project.json [hunk 3]
```diff
@@ -122,10 +222,10 @@
...
-            "dependsOn": []
+            "dependsOn": ["^build", "generate-typedoc"]
         },
         "serve": {
-            "executor": "@angular-builders/custom-webpack:dev-server",
+            "executor": "@angular-devkit/build-angular:dev-server",
```
[the remark this answers is not in the document]
Comment (mikerodonnell89) [reply apps/docs/project.json:228]: @platon-rov can you open PRs for this?

**Pull request merged**